source/logrepl/cdc.go

```diff
@@ -52,13 +59,19 @@ type CDCIterator struct {
 // NewCDCIterator sets up the subscription to a logical replication slot and
 // starts a goroutine that listens to events. The goroutine will keep running
 // until either the context is canceled or Teardown is called.
-func NewCDCIterator(ctx context.Context, conn *pgx.Conn, config Config) (*CDCIterator, error) {
+func NewCDCIterator(ctx context.Context, connPool *pgxpool.Pool, config Config) (*CDCIterator, error) {
 	i := &CDCIterator{
 		config:  config,
 		records: make(chan sdk.Record),
 	}
 
-	err := i.attachSubscription(ctx, conn)
+	conn, err := connPool.Acquire(ctx)
```

Comment:
I'd rather this be avoided, rather use the pool directly in the iterator for the following reasons:
1. The CDC iterator uses a single replication connection always
2. No need to explicitly release the connection, so safer for leaks.

Reply:
Wait, actually I just realized the way this flow is a bit odd. Since the pg repl connection doesn't actually need the pool but just the config of the connection, since the pool does not support replication connection types. 

So the only thing that will be extracted is the config, which makes the Acquire/Release an overhead since that connection  won't be used.

Reply:
We can supply the pool down into `attachSubscription` but we'll have to acquire a connection there anyway 🤷 We need a `TypeMap()` in line 179, as far as I can see we can only get it from a `pgx.Conn`. Note that this connection is released right away and is not actually used to fetch any data.

Reply:
No worries, to note is that establishing connection in Postgres is pretty expensive, since it needs to allocate all the workmem.

Reply:
Yeah, but do you see a way around it? We need that type map.

Reply:
I believe this is just `pgtype.NewMap()` which contains all the types.

Reply:
Though, I'm not sure how this may work with custom types which are not known by the map

Reply:
According by this https://pkg.go.dev/github.com/jackc/pgtype, new types will need to be explicitly registered so we are ok.

Reply:
Ok using `pgtype.NewMap` now 👍